### ISSUE TITLE
Deployment has not entered running state

### ISSUE THREAD

**Issue by AndreiOrmanji** [shuttle-hq/shuttle#569]
Description:
When I try to deploy, it can not be finished successfully.
Logs:
```
...
2023-01-07T16:49:40.580436879Z  INFO Entering built state

2023-01-07T16:49:40.614606048Z  INFO Entering loading state
2023-01-07T16:49:40.667410850Z  INFO shuttle_deployer::deployment::provisioner_factory: Provisioning a shared::postgres on the shuttle servers. This can take a while...
2023-01-07T16:49:40.714907239Z DEBUG {service.ready=true} tower::buffer::worker: processing request        
2023-01-07T16:49:40.931547381Z  INFO shuttle_deployer::deployment::provisioner_factory: Done provisioning database
Deployment has not entered the running state
```

Cargo.toml
```toml
[package]
name = "fool-game"
version = "0.1.0"
edition = "2021"
publish = false

[lib]

[workspace]
members = [
    ".",
    "entity",
    "migration"
]

[dependencies]
shuttle-service = { version = "0.8.0", features = ["web-actix-web"] }
# shuttle-aws-rds = { version = "0.8.0", features = ["postgres"] }
shuttle-shared-db = { version = "0.8.0", features = ["postgres"] }
# shuttle-secrets = { version = "0.8.0" }
sqlx = { version = "0.6.2", features = ["runtime-actix-native-tls", "postgres"] }
tracing ={ version = "*"}
entity = { path = "entity" }
migration = { path = "migration" } # depends on your needs
actix-web = "4.2"
anyhow = "1.0"
derive_more = "*"
dotenvy = "0.15"
chrono = { version = "0.4", features = ["serde"], optional = true }
chrono-tz ={ version = "0.8" , optional = true }
env_logger = "0.10"
log = "0.4"
playin-cards = "0.1"
rand = "0.8"
sea-orm = { version = "0.10", features = [
    "debug-print",
    "macros",
    # "runtime-tokio-native-tls",
    "runtime-actix-native-tls",
    # "sqlx-mysql",
    "sqlx-postgres",
    # "sqlx-sqlite",
    # "with-chrono",
    "with-time",
    "with-json",
    "with-rust_decimal",
    "with-uuid",
], default-features = false }
serde = { version = "1.0", features = ["derive"] }
serde_derive = "1.0"
serde_json = "1.0"
time = { version = "0.3", features = ["serde"], optional = true }
time-tz ={ version = "1.0" , optional = true }
tokio = { version = "1", features = ["rt-multi-thread"] }
tracing-subscriber = { version = "0.3", features = ["env-filter"] }

[features]
default = ["time"]
time = ["dep:time", "dep:time-tz"]
chrono = ["dep:chrono", "dep:chrono-tz"]

[dev-dependencies]
sea-orm = { version = "0.10", features = [
    "debug-print",
    "mock",
    "macros",
    "runtime-actix-native-tls",
    "sqlx-mysql",
    "sqlx-postgres",
    "sqlx-sqlite",
    "with-chrono",
    "with-time",
    "with-json",
    "with-rust_decimal",
    "with-uuid",
], default-features = false }

[profile.release]
debug = 1

```

**Comment by oddgrd** [comment]
Does it run locally? This error message, "Deployment has not entered the running state", usually means there is a panic on startup. I see you're using `dotenvy`, it might be the cause, since the `.env` files won't be included when your project is packaged and sent to our infrastructure. Could you try using our [secrets](https://docs.shuttle.rs/resources/shuttle-secrets) resource instead?

PS. we are planning to improve this error message soon, it's not great :smile:

**Comment by AndreiOrmanji** [comment]
It seems that `Deployment has not entered the running state` means that deploy was done successfully, but depoyed version of application failed to start or crashed immediately after start. I've found more information when I've tried to get logs of last deploy(thanks to Discord thread with almost same description). I think that the solution for this issue is the following steps:

1. get the list of deploys by running command `cargo shuttle deployment list`
2. identify deploy that failed (it usually has status `crashed`) and copy it's uuid.
3. use uuid copied from previous step to get full log of deploy and application by running `cargo shuttle logs <uuid>`, where `<uuid>` - uuid of deploy.
